pkg/core/resources/apis/mesh/external_service_helpers.go

```diff
@@ -0,0 +1,46 @@
+package mesh
+
+import (
+	"github.com/mitchellh/copystructure"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+)
+
+func (es *ExternalServiceResource) IsReachableFromZone(zone string) bool {
+	return es.Spec.Tags[mesh_proto.ZoneTag] == "" || es.Spec.Tags[mesh_proto.ZoneTag] == zone
+}
+
+func (esl *ExternalServiceResourceList) MarshalLog() interface{} {
+	maskedList := make([]*ExternalServiceResource, len(esl.Items))
+	for _, es := range esl.Items {
+		maskedList = append(maskedList, es.MarshalLog().(*ExternalServiceResource))
+	}
+	return ExternalServiceResourceList{
+		Items:      maskedList,
+		Pagination: esl.Pagination,
+	}
+}
+
+func (es *ExternalServiceResource) MarshalLog() interface{} {
+	c, err := copystructure.Copy(es)
```

Comment:
I think you can use protobuf copy instead (at least on the spec

Reply:
changed